learn/how-tos/aws-s3-connector.md

```diff
@@ -46,24 +46,23 @@ Uploads a new file object to the specified Amazon S3 bucket.
 Import statements required:
 ```Java
 import org.springframework.web.multipart.MultipartFile;
-import java.io.File;
+import java.io.InputStream;
+import java.io.ByteArrayInputStream;
 import java.io.IOException; 
 ```
 Source Code:
 ```Java
 public void uploadFile(MultipartFile file){
-    File fileObj=new File(file.getOriginalFilename());
-    try{
-        file.transferTo(fileObj);    
-    }catch(IOException e){
-        throw new RuntimeException("Exception occurred while converting to file: "+e);
+    try (InputStream inputStream = new ByteArrayInputStream(file.getBytes())) {
+        //Passing file object metadata as null. Pass custom metadata if any.
+        s3Connector.uploadFileToS3(inputStream,file.getOriginalFilename(), null);
```

Comment:
There is a way to get input steam from the MultipartFile object i.e file.getInputStream(). I do not think you need to use an intermediate ByteArrayInputStream.

Reply:
Updated with multipartfile.getInputStream()